core/src/main/java/com/linecorp/armeria/common/logging/TextLogFormatterBuilder.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.logging;
+
+import java.util.function.BiFunction;
+
+import com.linecorp.armeria.common.HttpHeaders;
+import com.linecorp.armeria.common.RequestContext;
+import com.linecorp.armeria.common.annotation.UnstableApi;
+
+/**
+ * A builder implementation for {@link TextLogFormatter}.
+ */
+@UnstableApi
+public final class TextLogFormatterBuilder extends AbstractLogFormatterBuilder<String> {
+
+    private static final BiFunction<RequestContext, HttpHeaders, String> DEFAULT_HEADERS_SANITIZER =
+            defaultSanitizer();
+    private static final BiFunction<RequestContext, Object, String> DEFAULT_CONTENT_SANITIZER =
+            defaultSanitizer();
+
+    private static <T, U> BiFunction<T, U, String> defaultSanitizer() {
+        return (first, second) -> second.toString();
+    }
+
+    TextLogFormatterBuilder() {}
+
+    @Override
+    public TextLogFormatterBuilder requestHeadersSanitizer(
+            BiFunction<? super RequestContext, ? super HttpHeaders, ? extends String> requestHeadersSanitizer) {
```

Comment:
Are you saying that we should add an example of sanitizer to each method as Javadoc?

Reply:
It would be useful. That being said, we may add it later after implementing the log writer feature that I mentioned on https://github.com/line/armeria/pull/4267#discussion_r1066696658. The API design could slightly change in the following PR.